lib/scholar/metrics/discounted_cumulative_gain.ex

```diff
@@ -0,0 +1,95 @@
+defmodule Scholar.Metrics.DiscountedCumulativeGain do
+  @moduledoc """
+  Discounted Cumulative Gain (DCG) is a measure of ranking quality.
+  It is based on the assumption that highly relevant documents appearing lower
+  in a search result list should be penalized as the graded relevance value is
+  reduced logarithmically proportional to the position of the result.
+  """
+
+  import Nx.Defn
+  import Scholar.Shared
+  require Nx
+
+  opts = [
+    k: [
+      default: nil,
```

Comment:
We usually do not use `nil` as a default because the value of a keyword list for entry that is not present in the keyword list is `nil`